src/distribution/ideal_soliton.rs

```diff
@@ -0,0 +1,248 @@
+use crate::distribution::Soliton;
+use crate::statistics::*;
+use crate::{Result, StatsError};
+use rand::distributions::Distribution;
+use rand::Rng;
+use std::f64;
+
+/// Implements the [Discrete
+/// Uniform](https://en.wikipedia.org/wiki/Discrete_uniform_distribution)
+/// distribution and an related ideal soliton of the discrete uniform distribuiton
+///
+/// # Examples
+///
+/// ```
+/// use statrs::distribution::{DiscreteUniform, Discrete};
+/// use statrs::statistics::Mean;
+/// use statrs::distribution::IdealSoliton;
+///
+/// let sol = IdealSoliton::new(5).unwrap();
+/// let n = DiscreteUniform::new(0, 5).unwrap();
+/// assert_eq!(n.mean(), 2.5);
+/// assert_eq!(n.pmf(3), 1.0 / 6.0);
+/// assert_eq!(sol.soliton(1), 1.0/5.0);
+/// ```
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct IdealSoliton {
+    min: i64,
+    max: i64,
+}
+
+impl IdealSoliton {
+    /// Constructs a new discrete uniform distribution with a minimum value
```

Comment:
There are seven doc strings referencing the discrete uniform distribution instead of the (ideal|robust) soliton distribution.

Reply:
Forgot to update this as had been done in robust soliton. :+1: